library/cmac.c

```diff
@@ -300,8 +299,10 @@ int mbedtls_cipher_cmac_update( mbedtls_cipher_context_t *ctx,
     /* If there is data left over that wasn't aligned to a block */
```

Comment:
`cmac_ctx->padding_flag` exists because of the special case of padding a 0 byte length input. After the initial initialisation call, unless we keep track of the number of bytes processed there's no way to tell if the input, it's blocked aligned, needs padding or not, unless we keep this flag.

After the initial block, you don't need padding any longer which is why we reset it after each block.